File: backend/vista-python-api/src/api/views/asset_categories.py
# SPDX-License-Identifier: Apache-2.0
# © Crown Copyright 2026. This work has been developed by the National Digital Twin Programme
# and is legally attributed to the Department for Business and Trade (UK) as the governing entity.

"""Views relating to Asset Categories."""

from rest_framework import viewsets

from api.models.asset_type import AssetCategory
from api.serializers import AssetCategorySerializer


class AssetCategoryViewSet(viewsets.ReadOnlyModelViewSet):
    """A ModelViewSet for the Asset Category model."""

    queryset = AssetCategory.objects.prefetch_related("sub_categories__asset_types").all()
    serializer_class = AssetCategorySerializer

File: frontend/src/api/asset-categories.ts
// SPDX-License-Identifier: Apache-2.0
// © Crown Copyright 2026. This work has been developed by the National Digital Twin Programme and is legally attributed to the Department for Business and Trade (UK) as the governing entity.

import config from '@/config/app-config';

export type AssetType = {
    id: string;
    name: string;
    icon?: string;
};

export type SubCategory = {
    id: string;
    name: string;
    assetTypes: AssetType[];
};

export type AssetCategory = {
    id: string;
    name: string;
    subCategories: SubCategory[];
};

export const fetchAssetCategories = async (): Promise<AssetCategory[]> => {
    const response = await fetch(`${config.services.apiBaseUrl}/assetcategories/`, {
        headers: { 'Content-Type': 'application/json' },
    });

    if (!response.ok) {
        throw new Error(`Failed to retrieve asset categories: ${response.statusText}`);
    }

    return await response.json();
};
